src/Symfony/Component/ExpressionLanguage/Parser.php

```diff
@@ -356,7 +364,8 @@ public function parsePostfixExpression(Node\Node $node)
                     throw new SyntaxError('Expected name.', $token->cursor, $this->stream->getExpression());
                 }
 
-                $arg = new Node\ConstantNode($token->value, true);
+                $isNullSafe = func_num_args() >= 2 && func_get_arg(1);
```

Comment:
actually, looking at the code more carefully, this new argument should not be needed. We can remove it if we don't recurse into parsePostfixExpression on L339. That'd be better.

Reply:
I agree! I smell a place of improvment over there .. let me think about it 🤔

Reply:
I just suggested a small refactoing of the `parsePostfixExpression` method which is BC-safe and at the same time avoids recursing on itself for the sake of applying the null-safety condition. The trick is to isolate the logic of object-access with the `.` punctuation (see the method `parseObjectAccessExpression`) since it will be the common re-usable logic in both scenarios. Same solution can be applied to the array-access logic with the `[` ponctuation (see the method `parseArrayAccessExpression`) to prepare the field for any future "special" mechanism to be seperately added their, for instance we can add the null-safe support for array-items-access as well if we choose too!! (a feature that even PHP8 itself does not support ^^).